crates/router/src/compatibility/stripe/errors.rs

```diff
@@ -425,6 +428,7 @@ impl From<errors::ApiErrorResponse> for StripeErrorCode {
             errors::ApiErrorResponse::CustomerNotFound => Self::CustomerNotFound,
             errors::ApiErrorResponse::PaymentNotFound => Self::PaymentNotFound,
             errors::ApiErrorResponse::PaymentMethodNotFound => Self::PaymentMethodNotFound,
+            errors::ApiErrorResponse::PaymentMethodTypeNotFound => Self::PaymentMethodTypeNotFound,
```

Comment:
If a field is missing, we raise `MissingRequiredField` error. Not found errors indicate that the resource was queried in the database and it could not find that particular row. So I think this error type is not needed.